picamera2/picamera2.py

```diff
@@ -522,26 +522,24 @@ def stream_configuration(self, name="main"):
         """Return the stream configuration for the named stream."""
         return self.camera_config[name]
 
-    def start_(self, controls={}):
-        """Start the camera system running."""
-        if self.camera_config is None:
-            raise RuntimeError("Camera has not been configured")
-        if self.started:
-            raise RuntimeError("Camera already started")
-        if self.camera.start(self.camera_config["controls"] | controls) >= 0:
```

Comment:
I think I'd prefer these error checks in `start_` rather than in `start`. There's a convention I've been trying to stick to (obviously, there's a bit of a problem in that no one else knows this, so sorry about that) where functions that end in `_` are guaranteed to be safe to call from the camera event loop, and the non-`_` versions aren't. Now, `start` and `start_` (and `configure` and `configure_`) are a bit different in that the camera can't be running, so they're actually the same. I was in several minds whether to create the two versions of each or not, but given that I did, I think that on balance I would prefer the old version. Thanks!

Reply:
Roger. Makes sense. To me, the '_' versions are for internal use only, so they should have functionality that other internal processes use. I'll switch it back. 

I would like to discuss the possibility of switching it to a preceding '_' so there is no confusion when someone is using an interactive IDE or if they are using the module as a source of information. That can be another day for another PR though.

Reply:
I think `_` for purely internal things sounds like a good idea. Sometimes it takes a while to learn exactly what really needs to be exposed and what doesn't, of course. But I agree it's something for another PR, my brain can only take so much at one time!